docs: fix linter errors for angular-compiler-options.md page (#47179) (#47618)
angular/angular: aio/content/guide/angular-compiler-options.md

When enabled, the `.js` output of `ngc` does not include any lazy-loaded template or style URLs.

For library projects created with the Angular CLI, the development configuration default is `true`.

<a id="enablelegacytemplate"></a>

### `enableLegacyTemplate`

> When `true`, enables use of the `<template>` element, deprecated in Angular 4.0, in favor of `<ng-template>`, to avoid colliding with the DOM element of the same name.

Reviewer: Thanks for the edit.
Now that we're well past Angular v14, we can just call it _deprecated_. The rest of the sentence after `<ng-template>` doesn't make any sense. `<template>` is an HTML element, while the original sentence makes it sound like `<ng-template>` is the DOM element.
```suggestion
When `true`, enables the deprecated `<template>` element in place of `<ng-template>`.
```

A flat module index `.d.ts` and `.js` is created with the given `flatModuleOutFile` name in the same location as the library index `.d.ts` file.

For example, if a library uses the `public_api.ts` file as the library index of the module, the `tsconfig.json` `files` field would be `["public_api.ts"]`.
The `flatModuleOutFile` option could then be set, for example, to `"index.js"`, which produces `index.d.ts` and `index.metadata.json` files.
The `module` field of the library's `package.json` would be `"index.js"` and the `typings` field would be `"index.d.ts"`.

### `fullTemplateTypeCheck`

> When `true`, recommended, enables the [binding expression validation](guide/aot-compiler#binding-expression-validation) phase of the template compiler, which uses TypeScript to verify binding expressions.

Reviewer: Just to make it a bit more readable.
```suggestion
When `true`, the recommended value, enables the [binding expression validation](guide/aot-compiler#binding-expression-validation) phase of the template compiler. This phase uses TypeScript to verify binding expressions.
```

For more information, see [Template type checking](guide/template-typecheck).

> Default is `false`, but when you use the Angular CLI command `ng new --strict`, it is set to `true` in the created project's configuration.

Reviewer: Perhaps:
```suggestion
Default is `false`, but when you use the Angular CLI command `ng new --strict`, it is set to `true` in the new project's configuration.
```

### `generateCodeForLibraries`

> When `true`, the default, generates factory files \(`.ngfactory.js` and `.ngstyle.js`\) for `.d.ts` files with a corresponding `.metadata.json` file.

Reviewer: Thanks for the edit.
In cases like this, it's easier to read when decomposed into

```suggestion
When `true`, creates factory files \(`.ngfactory.js` and `.ngstyle.js`\) for `.d.ts` files with a corresponding `.metadata.json` file. The default value is `true`.
```

When `false`, factory files are created only for `.ts` files.
Do this when using factory summaries.

### `preserveWhitespaces`

> When `false`, the default, removes blank text nodes from compiled templates, which results in smaller emitted template factory modules.

Reviewer: Thanks for the edit.
In cases like this, it's easier to read when decomposed into

```suggestion
When `false`, removes blank text nodes from compiled templates, which results in smaller emitted template factory modules. The default value is `false`.
```

Default is `false`.

The `.metadata.json` files contain information needed by the template compiler from a `.ts` file that is not included in the `.d.ts` file produced by the TypeScript compiler.

> This information includes, for example, the content of annotations, such as a component's template, which TypeScript emits to the `.js` file but not to the `.d.ts` file.

Reviewer: This sentence packs a lot of information such that it might be clearer if broken out like:

```suggestion
The `.metadata.json` files include Information such as annotations that TypeScript emits to the `.js` file but not to the `.d.ts` file. Annotations can include a component's template, for example.
```

For library projects created with the Angular CLI, the development configuration default is `true`.

### `strictMetadataEmit`

When `true`, reports an error to the `.metadata.json` file if `"skipMetadataEmit"` is `false`.
Default is `false`.
Use only when `"skipMetadataEmit"` is `false` and `"skipTemplateCodegen"` is `true`.

This option is intended to verify the `.metadata.json` files emitted for bundling with an `npm` package.
The validation is strict and can emit errors for metadata that would never produce an error when used by the template compiler.
You can choose to suppress the error emitted by this option for an exported symbol by including `@dynamic` in the comment documenting the symbol.

It is valid for `.metadata.json` files to contain errors.
The template compiler reports these errors if the metadata is used to determine the contents of an annotation.
The metadata collector cannot predict the symbols that are designed for use in an annotation. It preemptively includes error nodes in the metadata for the exported symbols.
The template compiler can then use the error nodes to report an error if these symbols are used.

> If the client of a library intends to use a symbol in an annotation, the template compiler does not normally report this. It gets reported once the client actually uses the symbol.

Reviewer: ```suggestion
If the client of a library intends to use a symbol in an annotation, the template compiler does not normally report this. It gets reported after the client actually uses the symbol.
```

This option allows detection of these errors during the build phase of the library and is used, for example, in producing Angular libraries themselves.

For library projects created with the Angular CLI, the development configuration default is `true`.

### `strictInjectionParameters`

> When `true`, recommended, reports an error for a supplied parameter whose injection type cannot be determined.

Reviewer: This and the following comment work together.
```suggestion
When `true`, reports an error for a supplied parameter whose injection type cannot be determined.
```

> When `false`, currently the default, constructor parameters of classes marked with `@Injectable` whose type cannot be resolved produce a warning.

Reviewer: ```suggestion
When `false`, constructor parameters of classes marked with `@Injectable` whose type cannot be resolved produce a warning.
The recommended value is `true`, but the default value is `false`.
```

When you use the Angular CLI command `ng new --strict`, it is set to `true` in the created project's configuration.

### `strictTemplates`

When `true`, enables [strict template type checking](guide/template-typecheck#strict-mode).

> Further strictness flags allow you to turn on and off specific types of strict template type checking.

Reviewer: ```suggestion
The strictness flags that this open enables allow you to turn on and off specific types of strict template type checking.
```

See [troubleshooting template errors](guide/template-typecheck#troubleshooting-template-errors).

> When you use the Angular CLI command `ng new --strict`, it is set to `true` in the created project's configuration.

Reviewer: ```suggestion
When you use the Angular CLI command `ng new --strict`, it is set to `true` in the new project's configuration.
```

Author: Thanks for the review. I've pushed a fixup with the suggested changes.

Reviewer: LGTM!
